### ISSUE TITLE
Make proxy manager required prop of core/App component

### ISSUE THREAD

**Issue by zachmullen** [Kitware/paraview-glance#107]
We get the following errors otherwise: 

```
[Vue warn]: Error in nextTick: "TypeError: Cannot read property 'onProxyRegistrationChange' of undefined"

found in

---> <Datasets> at node_modules/paraview-glance/src/components/core/Datasets/index.vue
       <VTabItem>
         <VTabsItems>
           <VTabs>
             <VNavigationDrawer>
               <VApp>
                 <DragAndDrop> at node_modules/paraview-glance/src/components/widgets/DragAndDrop/index.vue
                   <App> at node_modules/paraview-glance/src/components/core/App/index.vue
                     <Glance> at src/views/Glance.vue
                       <App> at src/App.vue
                         <Root>,TypeError: Cannot read property 'onProxyRegistrationChange' of undefined,[Vue warn]: Error in nextTick: "TypeError: Cannot read property 'onModified' of undefined"

found in

---> <GpuInformation> at node_modules/paraview-glance/src/components/widgets/GPUInformation/index.vue
       <GlobalSettings> at node_modules/paraview-glance/src/components/core/GlobalSettings/index.vue
         <VTabItem>
           <VTabsItems>
             <VTabs>
               <VNavigationDrawer>
                 <VApp>
                   <DragAndDrop> at node_modules/paraview-glance/src/components/widgets/DragAndDrop/index.vue
                     <App> at node_modules/paraview-glance/src/components/core/App/index.vue
                       <Glance> at src/views/Glance.vue
                         <App> at src/App.vue
                           <Root>,TypeError: Cannot read property 'onModified' of undefined,[Vue warn]: Error in nextTick: "TypeError: Cannot read property '$on' of undefined"

found in

---> <Notification> at node_modules/paraview-glance/src/components/core/Notification/index.vue
       <VApp>
         <DragAndDrop> at node_modules/paraview-glance/src/components/widgets/DragAndDrop/index.vue
           <App> at node_modules/paraview-glance/src/components/core/App/index.vue
             <Glance> at src/views/Glance.vue
               <App> at src/App.vue
                 <Root>,TypeError: Cannot read property '$on' of undefined,[Vue warn]: Error in nextTick: "TypeError: Cannot read property 'getViews' of undefined"

found in

---> <BrowserIssues> at node_modules/paraview-glance/src/components/core/BrowserIssues/index.vue
       <VApp>
         <DragAndDrop> at node_modules/paraview-glance/src/components/widgets/DragAndDrop/index.vue
           <App> at node_modules/paraview-glance/src/components/core/App/index.vue
             <Glance> at src/views/Glance.vue
               <App> at src/App.vue
                 <Root>,TypeError: Cannot read property 'getViews' of undefined,Script error.
```

**Comment by zachmullen** [comment]
For more information, right now `proxyManager` and `widgetManager` are expected to be provided using the `provide` directive. Unless there's a compelling reason to do it this way, I think we should just pass them as required props, since that is more idiomatic to vue.js.

It might also be nice to have some default values so that the `App` component could be used without needing to know a lot of internals of vtk.js, e.g. without having to write the following sort of code:

```
  const proxyConfiguration = proxyConfig || activeProxyConfig || Config.Proxy;

  const proxyManager = vtkProxyManager.newInstance({ proxyConfiguration });
  const renderListener = vtkListenerHelper.newInstance(
    proxyManager.autoAnimateViews,
    () =>
      [].concat(
        proxyManager.getSources(),
        proxyManager.getRepresentations(),
        proxyManager.getViews()
      )
  );

  proxyManager.onProxyRegistrationChange(renderListener.resetListeners);

  const widgetManager = vtkWidgetManager.newInstance({ proxyManager });
  widgetManager.registerWidgetGroup(Widgets.CROP, CropWidget);
```

**Comment by floryst** [comment]
Fixed with #129. proxyManager is part of a vuex store now.
